src/main/java/upeldev/com/github/upel3/controllers/AchievementsController.java

```diff
@@ -55,9 +57,16 @@ public String showAchievements(Model model,
         model.addAttribute("achievements", achievements);
         model.addAttribute("course", course);
 
-        if(course.getLecturers().contains(currentUser)) return "achievements_lecturer";
-        else return "error";
+        if(course.getLecturers().contains(currentUser) || currentUser.getRoles().contains(Role.ADMIN)){
+            return "achievements_lecturer";
+        }
+        else if(course.getEnrolledStudents().contains(currentUser)){
+            List<StudentAchievement> achieved = studentAchievementService.findAllByUserAndCourse(currentUser, course);
```

Comment:
Filter achieved list by isAchieved field - now if the achievement isn't satisfied, it still gets displayed.

Reply:
Just added `th:if="${achievement.isAchieved()}"`, which kinda does the same thing ultimately